src/main/java/com/serjltt/moshi/adapters/SerializeNullsJsonAdapter.java

```diff
@@ -28,39 +28,19 @@
 /**
  * {@linkplain JsonAdapter} that will serialize {@code T} even if the passed value is {@code null}.
  */
-public final class SerializeNullsJsonAdapter<T> extends JsonAdapter<T> {
+public final class SerializeNullsJsonAdapter {
   public static final JsonAdapter.Factory FACTORY = new JsonAdapter.Factory() {
     @Override public JsonAdapter<?> create(Type type, Set<? extends Annotation> annotations,
         Moshi moshi) {
       Set<? extends Annotation> nextAnnotations =
           Types.nextAnnotations(annotations, SerializeNulls.class);
       if (nextAnnotations == null) return null;
 
-      return new SerializeNullsJsonAdapter<>(moshi.adapter(type, nextAnnotations));
+      return moshi.adapter(type, nextAnnotations).serializeNulls();
     }
   };
 
-  private final JsonAdapter<T> delegate;
-
-  SerializeNullsJsonAdapter(JsonAdapter<T> delegate) {
-    this.delegate = delegate;
-  }
-
-  @Override public T fromJson(JsonReader reader) throws IOException {
-    return delegate.fromJson(reader);
-  }
-
-  @Override public void toJson(JsonWriter writer, T value) throws IOException {
-    boolean oldSerializeNulls = writer.getSerializeNulls();
-    writer.setSerializeNulls(true);
-    try {
-      delegate.toJson(writer, value);
-    } finally {
-      writer.setSerializeNulls(oldSerializeNulls);
-    }
-  }
-
-  @Override public String toString() {
-    return delegate + ".serializeNulls()";
+  private SerializeNullsJsonAdapter() {
+    throw new AssertionError("No instances.");
```

Comment:
I'm going to move all factories to their respective annotations and delete this.